platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -246,19 +261,26 @@ def get_firmware_version(self):
         Returns:
             A string containing the firmware version of the component
         """
-        cpld_version_file_list = glob(self.CPLD_VERSION_FILE_PATTERN)
-        cpld_version = ''
-        if cpld_version_file_list:
-            cpld_version_file_list.sort()
-            for version_file in cpld_version_file_list:
-                version = self._read_generic_file(version_file, self.CPLD_VERSION_MAX_LENGTH)
-                if cpld_version:
-                    cpld_version += '.'
-                cpld_version += version.rstrip('\n')
-        else:
-            raise RuntimeError("Failed to get CPLD version files by matching {}".format(self.CPLD_VERSION_FILE_PATTERN))
 
-        return cpld_version
+        part_number_file = self.CPLD_PART_NUMBER_FILE.format(self.idx)
+        version_file = self.CPLD_VERSION_FILE.format(self.idx)
+        version_minor_file = self.CPLD_VERSION_MINOR_FILE.format(self.idx)
+
+        part_number = self._read_generic_file(part_number_file, self.CPLD_PART_NUMBER_MAX_LENGTH, True)
+        version = self._read_generic_file(version_file, self.CPLD_VERSION_MAX_LENGTH)
+        version_minor = self._read_generic_file(version_minor_file, self.CPLD_VERSION_MINOR_MAX_LENGTH, True)
+
+        if part_number is None:
+            part_number = ZERO
```

Comment:
Is it better to use a more meaningful name here? Like `DEFAULT_PART_NUMBER` and `DEFAULT_VERSION_MINOR`?

Reply:
@stephenxs fixed